CMakeLists.txt

```diff
@@ -466,6 +466,14 @@ configure_file(contrib/filter-lcov.py filter-lcov.py COPYONLY)
 # Don't allow extended (non-ASCII) symbols in identifiers. This is easier for code review.
 try_append_cxx_flags("-fno-extended-identifiers" TARGET core_interface SKIP_LINK)
 
+# Make builds location independent. This helps to:
+#  1. Unify the -logsourcelocations format.
+#  2. Improve ccache efficiency across different directories.
```

Comment:
I also tried testing. It doesn't seem to work, because the `-I` argument passed to the compiler contains the the full `path/to/the/src` directory, which means the compiler invocations differ, and ccache wont hit?